xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -496,60 +607,196 @@ private <T extends ResourceUpdate> void handleResourceUpdate(
       // For State of the World services, notify watchers when their watched resource is missing
       // from the ADS update. Note that we can only do this if the resource update is coming from
       // the same xDS server that the ResourceSubscriber is subscribed to.
-      if (subscriber.serverInfo.equals(args.serverInfo)) {
+      if (controlPlaneClient.equals(subscriber.controlPlaneClient)) {
         subscriber.onAbsent(processingTracker);
       }
     }
   }
 
+  private void shutdownLowerPriorityCpcs(ControlPlaneClient activatedCpc) {
+    // For each authority, remove any control plane clients, with lower priority than the activated
+    // one, from activeCpClients storing them all in cpcsToSutdown.
+    List<String> authsForServerInfo = getAuthoritiesForServerInfo(activatedCpc.getServerInfo());
+    Set<ControlPlaneClient> cpcsToSutdown = new HashSet<>();
+    for (String authority : authsForServerInfo) {
+      List<ControlPlaneClient> cpcsForAuth = activeCpClients.get(authority);
+      int index = cpcsForAuth.indexOf(activatedCpc);
+      if (index > -1 && index < cpcsForAuth.size() - 1) {
+        cpcsToSutdown.addAll(cpcsForAuth.subList(index + 1, cpcsForAuth.size()));
+        for (int i = cpcsForAuth.size() - 1; i > index; i--) {
+          cpcsForAuth.remove(i);
+        }
+      }
+    }
+
+    // Shutdown any lower priority control plane clients identified above that aren't still being
```

Comment:
I don't think it is just about shutting down the lower priorities. We need to unsubscribe resources. That unsubscription will happen already by removing the cpc from `activeCpClients`, but we also need to call `adjustResourceSubscription()` for each removed client.

So I think this will become an if/else, where we either call shutdown() or adjustResourceSubscription() for each client.

Reply:
The subscriber still exists. But because a ControlPlaneClient will no longer be used for an authority, the resources for that authority are unsubscribed from the perspective of the ControlPlaneClient.

Reply:
Done